### PR TITLE
Supported services port

### PR DESCRIPTION
**Pull request by matt001k**
This PR is opened to show the changes made to source files to make them C compatible and bm_core compatible. Next PR will be the migration of these files to `bm_core`.

### CODE REVIEW

File: src/lib/debug/debug_bm_service.cpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ static const CLI_Command_Definition_t cmdBmService = {
     "bmsrv:\n"
     " * bmsrv req <service> <data> <timeout s>\n"
     " * bmsrv req sysinfo <node id> <timeout s>\n"
-    " * bmsrv req cfgmap <node id> <1(hw)/2(sys)/3(usr)> <timeout s>\n",
+    " * bmsrv req cfgmap <node id> <1(sys)/2(hw)/3(usr)> <timeout s>\n",
```
Comment (matt001k) [inline src/lib/debug/debug_bm_service.cpp:30]: This did not align with what is actually happening when running this command...

File: src/lib/sys/configuration.cpp
Comment (matt001k) [inline src/lib/sys/configuration.cpp:1]: This API was moved to bm_core.

File: test/src/common/configuration_ut.cpp
Comment (matt001k) [inline test/src/common/configuration_ut.cpp:1]: This unit test was moved to bm_core

File: src/lib/debug/debug_bm_service.cpp (hunk 1 above)
Comment (towynlin) [reply src/lib/debug/debug_bm_service.cpp:30]: Good eye!

**Review by towynlin — APPROVED** [review]

**Pull request merged**